### PR TITLE
refactor(qr-code): improve a11y attributes handling in QRCode component

### PR DESCRIPTION
**Pull request by aojunhao123**
### 🤔 This is a ...

- [x] 🛠 Refactoring
- [x] 🤖 TypeScript definition improvement
- [x] ✅ Test Case

### 🔗 Related Issues

N/A

#### Background
When passing props to QRCode's internal components (QRCodeCanvas/QRCodeSVG), accessibility-related attributes were not being properly forwarded, which could affect the component's accessibility support.

#### Solution
- Created a new utility function `pickA11yProps` to extract accessibility-related attributes
- Utilized existing `HTMLAriaDataAttributes` type for better type safety
- Applied the extracted a11y props to QRCode's canvas/svg elements
- Maintained all aria-* attributes and common a11y props (role, tabIndex, alt, title)

The solution improves both accessibility support and code maintainability without introducing breaking changes.

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Refactor: Improve accessibility attributes handling in QRCode component. |
| 🇨🇳 Chinese | 重构：优化 QRCode 组件无障碍属性的处理方式。 |

### CODE REVIEW

**Comment by aojunhao123** [comment]
@li-jia-nan @zombieJ @afc163 大佬们有空的时候帮忙review一下😌

File: components/_util/a11y.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { HTMLAriaDataAttributes } from './aria-data-attrs';
+
+export const pickA11yProps = (props: Record<string, any>): Partial<HTMLAriaDataAttributes> => {
```
Comment (li-jia-nan) [inline components/_util/a11y.ts:3]: 这里用 string 会不会更好一点？

```suggestion
export const pickA11yProps = (props: Record<string, string>): Partial<HTMLAriaDataAttributes> => {
```

Comment (zombieJ) [reply components/_util/a11y.ts:3]: 感觉 `string | number` 更好一些？

File: components/_util/a11y.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import type { HTMLAriaDataAttributes } from './aria-data-attrs';
```
Comment (zombieJ) [inline components/_util/a11y.ts:1]: rc-util 有一个 `pickAttrs` 方法支持那 a11y 属性，不用自己拿。如果 rc-util 的不够，就拓展一下。

File: components/_util/a11y.ts (hunk 1 above)
Comment (aojunhao123) [reply components/_util/a11y.ts:3]: 有类型冲突，我再看看

Comment (aojunhao123) [reply components/_util/a11y.ts]: 引用rc-utils中的pickAttrs了

File: components/qr-code/index.tsx [hunk 3]
```diff
@@ -55,6 +58,7 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
     fgColor: color,
     style: { width: style?.width, height: style?.height },
     imageSettings: icon ? imageSettings : undefined,
+    ...a11yProps,
```
Comment (zombieJ) [inline components/qr-code/index.tsx:61]: 测试用例

File: components/qr-code/index.tsx [hunk 4]
```diff
@@ -41,12 +42,14 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
...
     y: undefined,
-    height: typeof iconSize === 'number' ? iconSize : iconSize?.height ?? 40,
-    width: typeof iconSize === 'number' ? iconSize : iconSize?.width ?? 40,
+    height: typeof iconSize === 'number' ? iconSize : (iconSize?.height ?? 40),
+    width: typeof iconSize === 'number' ? iconSize : (iconSize?.width ?? 40),
     excavate: true,
     crossOrigin: 'anonymous',
   };
 
+  const a11yProps = pickAttrs(rest, { aria: true, data: true });
```
Comment (zombieJ) [inline components/qr-code/index.tsx:51]: 这边为什么要挪下去？另外如果你挪下去的话，那 rest 里的 aria 什么的应该去掉。
另外，`data-` 这个我理解还是应该在 div 上的

**Review by zombieJ — APPROVED** [review]

**Pull request merged**